pkgs/top-level/python-packages.nix

```diff
@@ -7802,6 +7802,49 @@ let
     };
   };
 
+  # Special check phase for numpy and scipy, following the same set of steps:
+  # First "install" the package, then import what was installed, and call the
+  # .test() function, which will run the test suite.
+  numpyScipyCheckPhase = python: pkgName: ''
```

Comment:
I think it would be nice to have this code snippet defined in a separate file, i.e. `pkgs/development/python-modules/numpy-scipy-checkphase.nix`. Then the two expressions that use it could `import` it from there. It's just a cosmetic change, but it seems a little odd to have that large code snippet in the middle of a file that mostly defines packages otherwise.